src/main/java/slimeknights/mantle/client/ExtraHeartRenderHandler.java

```diff
@@ -211,10 +211,11 @@ private void renderCustomHearts(int xBasePos, int yBasePos, int potionOffset, in
         this.drawTexturedModalRect(xBasePos + 8 * i, yBasePos + y, 0 + 18 * heartIndex, potionOffset, 9, 9);
       }
       if(count % 2 == 1 && renderHearts < 10) {
+        int y = renderHearts + regenOffset == regen ? -2 : 0;
```

Comment:
Maybe rewrite this thing to y = getYRegenOffset(i, regenOffset) and also replace the upper if with that call to keep them unified.